index.js

```diff
@@ -18,7 +19,10 @@ function message (commitText) {
   } else {
     node.children.push(s)
   }
-  if (scanner.eof()) return node
+  if (scanner.eof()) {
+    node.position = { start, end: scanner.position() }
```

Comment:
Perhaps an an eventual refactor we could have `scanner` return a `Node` object, which like you suggest could have an `exit` method?